### PR TITLE
bash-completion: update to 2.12.0

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by i0ntempest** [comment]
I think I have dealt with each patch correctly, either regenerating or deleting. If not please let me know.

**Comment by pmetzger** [comment]
LGTM. We can always fix it up a bit more later; the nice thing about the completions is that if they're slightly broken in a corner case, no software actually breaks.

File: sysutils/bash-completion/files/patch-xine-add-more-formats.diff [hunk 1]
```diff
@@ -4,27 +4,23 @@ Origin: vendor
 Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/bash-completion/+bug/311525
 Bug-Debian: https://bugs.debian.org/496162
```
Comment (raimue) [inline sysutils/bash-completion/files/patch-xine-add-more-formats.diff:5]: If this patch is no longer the version provided by these Debian/Ubuntu, we should delete/update the references. At least from the Debian source, I could not yet find an updated patch for 2.12.

https://salsa.debian.org/debian/bash-completion/-/blob/master/debian/patches/04-xine_add_more_formats.patch?ref_type=heads

File: sysutils/bash-completion/files/patch-xpdf-support-compressed-pdf.diff [hunk 2]
```diff
@@ -3,19 +3,16 @@ Subject: xpdf in Debian also supports compressed PDFs
 Origin: vendor, https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=702498
 Bug-Debian: https://bugs.debian.org/702498
```
Comment (raimue) [inline sysutils/bash-completion/files/patch-xpdf-support-compressed-pdf.diff:4]: Same here, Debian has not re-rolled this patch for 2.12 yet.

https://salsa.debian.org/debian/bash-completion/-/blob/master/debian/patches/06-xpdf_support_compressed_pdf.patch?ref_type=heads

**Review by raimue — COMMENTED** [review]
Thanks for the update! I would suggest we make it clear that these patches are no longer taken verbatim from another distribution, but have been updated for this port update. Otherwise looks good to me. :+1:

**Pull request merged**